### PR TITLE
Use kubevirtci instead of dind

### PR DESCRIPTION
**Pull request by RamLavi**
Currently, kubemacpool project includes both dind and kubevirtci. The usage of both of these is unfortunately quite messy. Travis is using dind, while the Makefile includes untested kubevirtci.

Let's align kubemacpool with our other projects and use kubevirtci only. By doing this switch, we will obtain unified way to run tests on kubemacpool. It will make maintenance and integration to prow easier.

Signed-off-by: Ram Lavi <ralavi@redhat.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -14,6 +14,14 @@ export PATH := $(GOBIN):$(PATH)
 GOFMT := $(GOBIN)/gofmt
 GO := $(GOBIN)/go
 
+CLUSTER_DIR ?= cluster
```
Comment (phoracek) [inline Makefile]: We don't need this change. All the scripts are already in the ./cluster/ directory. Is there any reason to parametrize those?

File: Makefile [hunk 2]
```diff
@@ -14,6 +14,14 @@ export PATH := $(GOBIN):$(PATH)
...
+KUBECONFIG ?= kubevirtci/_ci-configs/$(KUBEVIRT_PROVIDER)/.kubeconfig
+export KUBECTL ?= $(CLUSTER_DIR)/kubectl.sh
```
Comment (phoracek) [inline Makefile]: This is the only one I would keep.

File: Makefile [hunk 3]
```diff
@@ -78,19 +86,16 @@ docker-builder:
...
-	./cluster/up.sh
-
-cluster-down:
-	./cluster/down.sh
+cluster-up: $(CLUSTER_UP)
+	$(CLUSTER_UP)
 
-cluster-sync:
-	./cluster/sync.sh
+cluster-down: $(CLUSTER_DOWN)
```
Comment (phoracek) [inline Makefile]: No need to specify the dependency. cluster/ directory is part of the repo.

File: Makefile [hunk 4]
```diff
@@ -78,19 +86,16 @@ docker-builder:
...
+	$(CLUSTER_DOWN)
 
-deploy-test-cluster:
-	./hack/deploy-test-cluster.sh
+cluster-sync: $(CLUSTER_SYNC)
```
Comment (phoracek) [inline Makefile]: ditto

File: Makefile [hunk 5]
```diff
@@ -29,11 +37,11 @@ functest: $(GO)
 	GO=$(GO) ./hack/functest.sh
 
 # Deploy controller in the configured Kubernetes cluster in ~/.kube/config
-deploy: generate-deploy
-	kubectl apply -f config/test/kubemacpool.yaml
+deploy: $(KUBECTL) generate-deploy
```
Comment (qinqon) [inline Makefile]: $(KUBECTL) is not longer a dependency

File: Makefile [hunk 6]
```diff
@@ -29,11 +37,11 @@ functest: $(GO)
...
+	$(KUBECTL) apply -f config/test/kubemacpool.yaml
 
-deploy-test: generate-test
-	kubectl apply -f config/test/kubemacpool.yaml
+deploy-test: $(KUBECTL) generate-test
```
Comment (qinqon) [inline Makefile]: ditto

File: Makefile [hunk 7]
```diff
@@ -78,19 +86,16 @@ docker-builder:
 docker-push:
 	docker push ${REGISTRY}/${IMG}:${IMAGE_TAG}
 
-cluster-up:
-	./cluster/up.sh
-
-cluster-down:
-	./cluster/down.sh
+cluster-up: $(CLUSTER_UP)
```
Comment (qinqon) [inline Makefile]: dep not needed

File: Makefile (hunk 1 above)
Comment (RamLavi) [reply Makefile]: just because it's used more than once, then I added it for easier future changes.

**Pull request merged**